cpp/src/traversal/extract_bfs_paths_impl.cuh

```diff
@@ -0,0 +1,378 @@
+
+      std::forward_as_tuple(std::tie(tmp_frontier, original_position, original_rank), std::ignore) =
+        groupby_gpuid_and_shuffle_values(
+          handle.get_comms(),
+          tuple_begin,
+          tuple_begin + tmp_frontier.size(),
+          [key_func =
+             detail::mg_partition_vertices<vertex_t>{
+               handle.get_comms().get_rank(),
+               d_vertex_partition_offsets.begin(),
+               d_vertex_partition_offsets.end()}] __device__(auto val) {
+            return key_func(thrust::get<0>(val));
+          },
+          handle.get_stream());
+
+      tuple_begin = thrust::make_zip_iterator(
+        thrust::make_tuple(tmp_frontier.begin(), original_position.begin(), original_rank.begin()));
+
+      thrust::transform(
+        handle.get_thrust_policy(),
+        tuple_begin,
+        tuple_begin + tmp_frontier.size(),
+        tuple_begin,
+        detail::mg_lookup_predecessor<vertex_t>{graph_view.get_local_vertex_first(), predecessors});
+
+      std::forward_as_tuple(std::tie(tmp_frontier, original_position, original_rank), std::ignore) =
+        groupby_gpuid_and_shuffle_values(
+          handle.get_comms(),
+          tuple_begin,
+          tuple_begin + tmp_frontier.size(),
+          [] __device__(auto val) { return thrust::get<2>(val); },
+          handle.get_stream());
+
+      thrust::for_each_n(handle.get_thrust_policy(),
+                         thrust::make_zip_iterator(
+                           thrust::make_tuple(tmp_frontier.begin(), original_position.begin())),
+                         tmp_frontier.size(),
+                         [d_current_frontier = current_frontier.data()] __device__(auto tuple) {
+                           auto v        = thrust::get<0>(tuple);
+                           auto position = thrust::get<1>(tuple);
+
+                           d_current_frontier[position] = v;
+                         });
+    } else {
+      thrust::transform(
+        handle.get_thrust_policy(),
+        current_frontier.begin(),
+        current_frontier.end(),
+        current_frontier.data(),
+        detail::sg_lookup_predecessor<vertex_t>{graph_view.get_local_vertex_first(), predecessors});
+    }
+
+    std::tie(current_frontier, current_position) = detail::shrink_extraction_list(
+      handle, std::move(current_frontier), std::move(current_position));
+
+    thrust::for_each_n(handle.get_thrust_policy(),
+                       thrust::make_zip_iterator(
+                         thrust::make_tuple(current_frontier.begin(), current_position.begin())),
+                       current_frontier.size(),
+                       detail::update_paths<vertex_t>{paths, invalid_vertex});
+
+    count = multi_gpu ? cugraph::host_scalar_allreduce(handle.get_comms(),
+                                                       current_frontier.size(),
+                                                       raft::comms::op_t::MAX,
```

Comment:
Shouldn't this better be SUM reduction? (to minimize the difference in SG & MG logics)

Reply:
Same as above.

Reply:
Same fix as above